File: LeetCode_13/Solution_ZhongJing.py
"""
日期：2023-08-04
作者：仲景
"""


class Solution:
    def romanToInt(self, s: str) -> int:
        # 创建一个map
        romanMap = {
            "I": 1,
            "V": 5,
            "X": 10,
            "L": 50,
            "C": 100,
            "D": 500,
            "M": 1000,
        }
        
        # 遍历str
        res = 0
        for i in range(len(s) - 1):
            # 如果当前符号比后一个小，说明是后一个-当前
            if romanMap[s[i]] < romanMap[s[i + 1]]:
                res -= romanMap[s[i]]
            # 如果当前符号>=后一个，说明是+当前
            else:
                res += romanMap[s[i]]
        # 最后一个无论如何都是+
        res += romanMap[s[len(s) - 1]]
        
        return res

File: LeetCode_13/LeetCode_13_03/Solution_ZhongJing.kt
package LeetCode_13.LeetCode_13_03

/**
 * 日期：2023-07-02 <br>
 * 作者：仲景
 */
class Solution {
    fun romanToInt(s: String): Int {
        var numMap = mapOf<Char, Int>(
            'I' to 1,
            'V' to 5,
            'X' to 10,
            'L' to 50,
            'C' to 100,
            'D' to 500,
            'M' to 1000,
        )

        var res = 0

        // 遍历字符串
        for (i in 0 until s.length - 1) {
            if (numMap[s[i]]!! < numMap[s[i + 1]]!!) {
                res -= numMap[s[i]]!!
            } else {
                res += numMap[s[i]]!!
            }
        }

        res += numMap[s[s.length - 1]]!!

        return res
    }
}
